app/models/participant_declaration.rb

```diff
@@ -50,8 +50,22 @@ def refresh_payability!
         participant_profile: participant_profile,
         participant_declaration: self,
         payable: currently_payable,
+        voided: current_profile_declaration&.voided || false,
       )
       reload
     end
   end
+
+  def void!
+    return if current_profile_declaration&.voided
+
+    # TODO: Prevent voiding a processed declaration - that requires clawbacks
+    ProfileDeclaration.create!(
```

Comment:
I didn't expect us to create a new record, but to update an existing one with `voided_at: Time.now.current` or similar. What's the motivation for this approach?

Reply:
Hmmm, I think I like keeping a single `voided_at` date. I didn't think too hard on it, the general feeling I had was "well, we need to keep track of when it happened, and those things should live in profile declarations".